translate/src/core/utils/hooks/useOnDiscard.ts

```diff
@@ -1,29 +1,21 @@
-import * as React from 'react';
+import { useEffect } from 'react';
 
 export default function useOnDiscard(
-  ref: { current: null | React.ElementRef<any> },
+  ref: React.RefObject<unknown>,
```

Comment:
> This is weird, as `React` isn't actually defined anymore?

Huh, yeah, that is a bit weird. But TS seems ok with it?

> Could this just create the ref itself and return it? Then it could be typed to an `HTMLElement | null` without having to type that everywhere this is used.

Yeah, that would work. Haven't really touched the core structural approach here, but got a gut sense that some different solution to the fundamental click-elsewhere-to-close action would be nicer.